server/src/main/java/org/opensearch/search/SearchHit.java

```diff
@@ -203,10 +204,14 @@ public SearchHit(StreamInput in) throws IOException {
         sortValues = new SearchSortValues(in);
 
         size = in.readVInt();
-        if (size > 0) {
-            matchedQueries = new String[size];
+        if (in.getVersion().onOrAfter(Version.V_2_12_0)) {
```

Comment:
We sadly have to go 2 steps, otherwise BWC tests with mixed clusters will fail:

```suggestion
        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Reply:
done